src/ui/Frame.cpp

```diff
@@ -664,79 +668,93 @@ Frame::result_t Frame::process(SDL_Rect _size, SDL_Rect _actualSize, const std::
 		}
 	}
 
+	// process frames
+	{
+		for (int i = frames.size() - 1; i >= 0; --i) {
+			Frame* frame = frames[i];
+			result_t frameResult = frame->process(_size, actualSize, selectedWidgets, usable);
+			usable = result.usable = frameResult.usable;
+			if (!frameResult.removed) {
+				if (frameResult.tooltip != nullptr) {
+					result = frameResult;
+				}
+			} else {
+				delete frame;
+				frames.erase(frames.begin() + i);
+			}
+		}
+	}
+
 	// scroll with right stick
-	if (allowScrolling && allowScrollBinds) {
+	if (usable && allowScrolling && allowScrollBinds) {
 		Input& input = Input::inputs[owner];
 
 		// x scroll
 		if (this->actualSize.w > size.w) {
 			if (input.binary("MenuScrollRight")) {
 				this->actualSize.x += std::min(this->actualSize.x + 5, this->actualSize.w - _size.w);
+				usable = result.usable = false;
 			}
 			else if (input.binary("MenuScrollLeft")) {
 				this->actualSize.x -= std::max(this->actualSize.x - 5, 0);
+				usable = result.usable = false;
 			}
 		}
 
 		// y scroll
 		if (this->actualSize.h > size.h) {
 			if (input.binary("MenuScrollDown")) {
 				this->actualSize.y = std::min(this->actualSize.y + 5, this->actualSize.h - _size.h);
+				usable = result.usable = false;
 			}
 			else if (input.binary("MenuScrollUp")) {
 				this->actualSize.y = std::max(this->actualSize.y - 5, 0);
+				usable = result.usable = false;
 			}
 		}
 	}
 
 	// scroll with mouse wheel
-	if (parent != nullptr && !hollow && rectContainsPoint(fullSize, omousex, omousey) && usable && allowScrolling && allowScrollBinds) {
-		// x scroll with mouse wheel
-		if (this->actualSize.w > size.w) {
-			if (this->actualSize.h <= size.h) {
-				if (mousestatus[SDL_BUTTON_WHEELDOWN]) {
-					mousestatus[SDL_BUTTON_WHEELDOWN] = 0;
-					this->actualSize.x += std::min(entrySize * 4, size.w);
-					usable = result.usable = false;
-				} else if (mousestatus[SDL_BUTTON_WHEELUP]) {
-					mousestatus[SDL_BUTTON_WHEELUP] = 0;
-					this->actualSize.x -= std::min(entrySize * 4, size.w);
-					usable = result.usable = false;
-				}
-			}
+	if (parent != nullptr && !hollow && rectContainsPoint(fullSize, omousex, omousey) && usable) {
```

Comment:
`allowScrolling && allowScrollBinds` was removed, this consumes the mouse wheel when any frame is open and doesn't check if scrolling is needed